src/React.Sample.Mvc4/Controllers/HomeController.cs

```diff
@@ -86,10 +88,25 @@ public ActionResult Comments(int page)
 		    var comments = _comments.Skip((page - 1) * COMMENTS_PER_PAGE).Take(COMMENTS_PER_PAGE);
 			var hasMore = page * COMMENTS_PER_PAGE < _comments.Count;
 
-		    return Json(new {
-				comments = comments,
-				hasMore = hasMore
-			}, JsonRequestBehavior.AllowGet);
+	        if (ControllerContext.HttpContext.Request.ContentType == "application-json")
```

Comment:
I would prefer if there were two separate actions - One that returns JSON, and one that returns HTML. The `loadMoreClicked` function in JS can hit the JSON endpoint, while the <a>'s href can be the HTML one.

Alternatively, switching by content-type is fine, but you **must** send a `Vary: Content-Type` header to prevent proxies from incorrectly caching it. I think the OutputCache attribute would let you do that.